### PR TITLE
CI: use Go 1.23; go.mod: go 1.22

### PR DESCRIPTION
**Pull request by AkihiroSuda**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/lima-vm/lima
 
-go 1.21
+go 1.22.0
```
Comment (alexandear) [inline go.mod:3]: Could we use `go 1.22`?

Comment (AkihiroSuda) [reply go.mod:3]: No, the syntax was changed to require the patch version too.

Comment (alexandear) [reply go.mod:3]: Should we use the latest available patch version?
```suggestion
go 1.22.6
```

Comment (AkihiroSuda) [reply go.mod:3]: I don't think it is necessary.
It is user's responsibility to update Go.

Comment (AkihiroSuda) [reply go.mod:3]: Can we merge?

Comment (AkihiroSuda) [reply go.mod]: `go 1.22` seems to work with the current toolchains

Comment (AkihiroSuda) [reply go.mod]: CI is passing with `go 1.22`, so let me merge this

**Pull request merged**